source/agora/utils/InetUtils.d

```diff
@@ -41,6 +51,17 @@ public enum HostType : ubyte
     Domain,
 }
 
+struct Service {
+	enum ipify = Service("api.ipify.org", "/");							/// https://www.ipify.org/
+	enum plain_text_ip = Service("plain-text-ip.com", "/");				/// http://about.plain-text-ip.com/
+	enum icanhazip = Service("icanhazip.com", "/");						/// http://icanhazip.com/
+	enum whatismyipaddress = Service("bot.whatismyipaddress.com", "/");	/// https://whatismyipaddress.com/api
+	enum amazonws = Service("checkip.amazonaws.com", "/");				/// http://checkip.amazonaws.com/
```

Comment:
Is the plan to try each of these services until a valid `ip` is returned ?

Reply:
Actually not but we can. I will discuss it with @Geod24.